service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java

```diff
@@ -0,0 +1,215 @@
+        } else {
+            previousHearingBooking = caseData.getMostUrgentHearingBookingAfter(time.now());
+        }
+
+        return hearingVenueLookUpService.getHearingVenue(previousHearingBooking);
+    }
+
+    public Map<String, Object> populatePreviousVenueFields(CaseData caseData) {
+        Map<String, Object> data = new HashMap<>();
+        HearingVenue previousHearingVenue = getPreviousHearingVenue(caseData);
+
+        Address customAddress = "OTHER".equals(previousHearingVenue.getHearingVenueId())
+            ? previousHearingVenue.getAddress() : null;
+
+        data.put("previousHearingVenue",
+            PreviousHearingVenue.builder()
+                .previousVenue(hearingVenueLookUpService.buildHearingVenue(previousHearingVenue))
+                .newVenueCustomAddress(customAddress)
+                .build());
+
+        return data;
+    }
+
+    public HearingBooking findHearingBooking(UUID id, List<Element<HearingBooking>> hearingBookings) {
+        Optional<Element<HearingBooking>> hearingBookingElement = findElement(id, hearingBookings);
+
+        if (hearingBookingElement.isPresent()) {
+            return hearingBookingElement.get().getValue();
+        }
+
+        return HearingBooking.builder().build();
+    }
+
+    public Map<String, Object> populateHearingCaseFields(HearingBooking hearingBooking) {
+        Map<String, Object> caseFields = new HashMap<>();
+
+        if (OTHER.equals(hearingBooking.getType())) {
+            caseFields.put("hearingTypeDetails", hearingBooking.getTypeDetails());
+        }
+
+        caseFields.put("hearingType", hearingBooking.getType());
+        caseFields.put("hearingStartDate", hearingBooking.getStartDate());
+        caseFields.put("hearingEndDate", hearingBooking.getEndDate());
+        caseFields.put("judgeAndLegalAdvisor", hearingBooking.getJudgeAndLegalAdvisor());
+
+        if (hearingBooking.getPreviousHearingVenue() == null
+            || hearingBooking.getPreviousHearingVenue().getPreviousVenue() == null) {
+            caseFields.put("hearingVenue", hearingBooking.getVenue());
+            caseFields.put("hearingVenueCustom", hearingBooking.getVenueCustomAddress());
+        } else {
+            caseFields.put("previousHearingVenue", hearingBooking.getPreviousHearingVenue());
+        }
+
+        return caseFields;
+    }
+
+    public void findAndSetPreviousVenueId(CaseData caseData) {
+        if (caseData.getHearingDetails() != null && caseData.getPreviousHearingVenue() != null
+            && caseData.getPreviousHearingVenue().getUsePreviousVenue().equals("Yes")) {
+
+            PreviousHearingVenue previousVenueForEditedHearing = caseData.getPreviousHearingVenue();
+
+            caseData.setPreviousVenueId(hearingVenueLookUpService.getVenueId(
+                previousVenueForEditedHearing.getPreviousVenue()));
+        }
+    }
+
+    public HearingBooking buildHearingBooking(CaseData caseData) {
```

Comment:
If you do the judge and legal advisor migration stuff in these private methods then you wouldn't need to get the selected judge when building the notice of hearing I think. And it would mean that the tabs look more consistent with their current design.